packages/client/hmi-client/src/model-representation/service.ts

```diff
@@ -145,3 +147,128 @@ export const getModelRenderer = (
 		dragSelector: 'no-drag'
 	});
 };
+
+/**
+ * Returns the model parameters based on the model type.
+ * @param {Model} model - The model object.
+ * @returns {ModelParameter[]} - The model parameters.
+ */
+export function getModelParameters(model: Model): ModelParameter[] {
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			return model.model?.parameters ?? [];
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			return model.semantics?.ode?.parameters ?? [];
+	}
+}
+
+/**
+ * Returns the model parameter with the specified ID.
+ * @param {Model} model - The model object.
+ * @param {string} parameterId - The ID of the parameter.
+ * @returns {ModelParameter | null} - The model parameter or null if not found.
+ */
+export function getParameter(model: Model, parameterId: string): ModelParameter | null {
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			return model.model?.parameters.find((p) => p.id === parameterId);
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			return model.semantics?.ode?.parameters?.find((p) => p.id === parameterId) ?? null;
+	}
+}
```

Comment:
For `getParameter` and `getInitial` - you are distinguish between null and undefined, it would be good if we just stick to one or the other.

Also it would be nice if we have more consistent naming, e.g. you have:
- getModelParameters/getParameter
- getModelInitials/getInitial